themes/default/content/docs/concepts/inputs-outputs/_index.md

```diff
@@ -13,30 +13,346 @@ aliases:
+{{< chooser language "javascript,typescript,python,go,csharp,java" >}}
+
+{{% choosable language javascript %}}
+
+```javascript
+// TODO
+```
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+const myId = new random.RandomId("mine", {
+    byteLength: 8, // byte_length is an input
+});
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+my_id = random.RandomId("mine",
+    byte_length=8 # byte_length is an input
+)
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+
+myId, err := random.NewRandomId(ctx, "mine", &random.RandomIdArgs{
+	ByteLength: pulumi.Int(8), // ByteLength is an input
+})
+if err != nil {
+	return err
+}
+
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var myId = new Random.RandomId("mine", new()
+{
+    ByteLength = 8, // ByteLength is an input
+});
+
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+var myId = new RandomId("mine", RandomIdArgs.builder()
+    .byteLength(8) // byteLength is an input
+    .build());
+
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+_Inputs_ are generally representations of the parameters to the underlying API call of any resource that Pulumi is managing.
+
+The simplest way to create a resource with its required _inputs_ is to use a _plain value_.
+
+{{< chooser language "javascript,typescript,python,go,csharp,java" >}}
+
+{{% choosable language javascript %}}
+
+```javascript
+const key = new tls.PrivateKey("my-private-key", {
+    algorithm: "ECDSA", // ECDSA is a plain value
+});
+
+```
+
+{{% /choosable %}}
+{{% choosable language typescript %}}
+
+```typescript
+const key = new tls.PrivateKey("my-private-key", {
+    algorithm: "ECDSA", // ECDSA is a plain value
+});
+
+```
+
+{{% /choosable %}}
+{{% choosable language python %}}
+
+```python
+key = tls.PrivateKey("my-private-key",
+  algorithm="ECDSA", # ECDSA is a plain value
+)
+
+```
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+```go
+
+key, err := tls.NewPrivateKey(ctx, "my-private-key", &tls.PrivateKeyArgs{
+	Algorithm:  pulumi.String("ECDSA"), // ECDSA is a plain value
+})
+if err != nil {
+	return err
+}
+
+
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+var key = new PrivateKey("my-private-key", new PrivateKeyArgs{
+    Algorithm = "ECDSA", // ECDSA is a plain value
+});
+```
+
+{{% /choosable %}}
+{{% choosable language java %}}
+
+```java
+// TODO
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+{{% notes %}}
+_Plain value_ in this document is used to describe a standard string, boolean, integer or other typed value or data structure in your language of choice. _Plain_ value_ is a way of differentiating these language specific values from Pulumi's asynchronous values.
+
+An {{< pulumi-output >}} value can resolve to a _plain value_. For more information, see [apply](#apply).
```

Comment:
This feels confusing to say this way - and as though it’s something that suggests you should try to wait for these values to resolve and pass that “by far the easiest” kind of resulting value. I think we want to intentionally avoid encouraging that mental model. 

Instead - we should encourage that you can pass outputs as inputs. And if you want to transform the value of an output, you can do so within an apply. But I think it is somewhat actively dangerous to encourage the promise like model of awaiting or resolving these values - as that’s exactly what gets folks in trouble when working with Pulumi outputs.

Reply:
Reading through the rest of this - I think we can just remove this paragraph. This concept is explained well in context later, after introducing outputs where it makes more sense.